experimenter/experimenter/experiments/api/v5/serializers.py

```diff
@@ -1490,14 +1490,47 @@ def _validate_feature_value_against_schema(
 
         return None
 
-    def _validate_with_fml(self, loader: NimbusFmlLoader, feature_id: str, obj):
-        if fml_errors := loader.get_fml_errors(obj, feature_id):
+    def _validate_with_fml(
+        self,
+        loader: NimbusFmlLoader,
+        feature_config,
+        blob: str,
+        schemas_in_range: NimbusFeatureConfig.VersionedSchemaRange,
+    ):
+        if schemas_in_range.unsupported_in_range:
             return [
-                f"{NimbusExperiment.ERROR_FML_VALIDATION}: "
-                f"{e.message} at line {e.line+1} column {e.col}"
-                for e in fml_errors
+                NimbusConstants.ERROR_FEATURE_CONFIG_UNSUPPORTED_IN_RANGE.format(
+                    feature_config=feature_config,
+                )
             ]
-        return None
+        errors = []
+
+        if (
+            len(schemas_in_range.schemas) == 1
+            and schemas_in_range.schemas[0].version is None
+        ):
+            versions = []
+        else:
+            versions = [schema.version for schema in schemas_in_range.schemas]
+
+        for version in versions:
+            for version in schemas_in_range.unsupported_versions:
+                errors.append(
+                    NimbusConstants.ERROR_FEATURE_CONFIG_UNSUPPORTED_IN_VERSION.format(
+                        feature_config=feature_config,
+                        version=version,
+                    )
+                )
+            if fml_errors := loader.get_fml_errors(blob, feature_config.slug, version):
+                errors.extend(
+                    [
+                        f"{NimbusExperiment.ERROR_FML_VALIDATION}: "
+                        f"{e.message} at line {e.line+1} column {e.col}"
+                        f"{f' at version {version}' if version is not None else ''}"
+                        for e in fml_errors
+                    ]
+                )
```

Comment:
The Python ternary is really unreadable IMO and nested formatting strings is confusing. Can we rewrite this as a regular loop, e.g.

```suggestion
                for error in fml_errors:
                    msg = (
                        f"{NimbusExperiment.ERROR_FML_VALIDATION}: "
                        f"{e.message} at line {e.line+1} column {e.col}"
                    )
                    if version:
                        msg = f"{msg} at version {version}"
                    
                    errors.append(msg)
```

Reply:
Ooooo that's MUCH nicer to read.